src/metabase/channel/email.clj

```diff
@@ -198,31 +198,53 @@
         (and (sequential? message) (every? map? message))
         (string? message)))]])
 
+;; TODO: this indirection is not needed, refactor to let all clients use the postal message directly
+(defn email-message->postal-message
+  "Converts a Metabase email message to a postal message. This is used internally by [[send-email!]] and
+  [[send-message!]]."
+  [{:keys [subject recipients message-type message bcc?] :as _email}]
+  (let [to-type (if bcc? :bcc :to)]
+    (merge
+     {:from    (if-let [from-name (email-from-name)]
+                 (str from-name " <" (email-from-address) ">")
+                 (email-from-address))
+      ;; FIXME: postal doesn't accept recipients if it's a set, need to fix this from upstream
+      to-type  (seq recipients)
+      :subject subject
+      :body    (case message-type
+                 :attachments message
+                 :text        message
+                 :html        [{:type    "text/html; charset=utf-8"
+                                :content message}])}
+     (when-let [reply-to (email-reply-to)]
+       {:reply-to reply-to}))))
+
 (defn send-message-or-throw!
   "Send an email to one or more `recipients`. Upon success, this returns the `message` that was just sent. This function
   does not catch and swallow thrown exceptions, it will bubble up. Should prefer to use [[send-email-retrying!]] unless
   the caller has its own retry logic."
-  [{:keys [subject recipients message-type message bcc?] :as _email}]
+  [email]
+  (try
+    (when-not (email-smtp-host)
+      (throw (ex-info (tru "SMTP host is not set.") {:cause :smtp-host-not-set})))
+    ;; Now send the email
+    (send-email! (smtp-settings) (email-message->postal-message email))
+    (catch Throwable e
+      (analytics/inc! :metabase-email/message-errors)
+      (when (not= :smtp-host-not-set (:cause (ex-data e)))
+        (throw e)))
+    (finally
+      (analytics/inc! :metabase-email/messages))))
+
+(defn send-postal-message-or-throw!
+  "Like [[send-message-or-throw!]] but accept a post email instead of our internal email data strucuture.
+  We should refactor all call sites to use this function to remove indirection."
+  [email]
   (try
     (when-not (email-smtp-host)
       (throw (ex-info (tru "SMTP host is not set.") {:cause :smtp-host-not-set})))
     ;; Now send the email
-    (let [to-type (if bcc? :bcc :to)]
-      (send-email! (smtp-settings)
-                   (merge
-                    {:from    (if-let [from-name (email-from-name)]
```

Comment:
don't even know why we have this weird transformation in the first place, might as well use the postal's datastructure right? renaming body to message doesn't make any different.